src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -65,6 +72,47 @@ private void AddExceptionDiagnostic(Diagnostic diagnostic)
         }
 
         /// <summary>
+        /// Returns diagnostics produced by diagnostic analyzers from analyzing a single document.
```

Comment:
Is this strictly diagnostics from a single document (you should probably use the term 'tree')? Consider the below cases:

1. If symbols/events from this tree were already completed by force completing some other tree, and the corresponding diagnostics dequeued, then those diagnostics will not be returned again by this invocation.
2. If force completing SemanticModel.GetDiagnostics for this tree, causes events to be generated for other trees, then `_driver.GetPartialDiagnosticsAsync` will even process those events and return diagnostics from it.

Add to the mix the complexity of non-local diagnostics that could be reported in this tree by compilation/compilation end actions, and we surely cannot guarantee what this comment intuitively states. Probably only thing we can guarantee is that no further local diagnostics can be generated for this tree after this invocation has been awaited?